R/query_index.R

```diff
@@ -63,15 +70,36 @@ query_index <- R6::R6Class(
     #'
     #' @return A new query_index object with the scoped index
     get_index_scoped = function() {
-      query_index$new(self$root, private$index_scoped)
+      query_index$new(self$root, private$index_scoped, private$depends)
+    },
+
+    #' @description
+    #' Get the ids of packets which this packet depends to a specified level
+    #'
+    #' @param id The id of the packet to get parents of
+    #' @param depth Depth of parents to get, `depth` 1 gets immediate parents
+    #' `depth` 2 gets parents and parents of parents, `depth` Inf will
+    #' recurse the whole tree to get all parents
+    #' @return The ids of the parents of this packet
+    get_packet_depends = function(id, depth) {
+      deps <- unique(private$get_all_packet_depends(id, depth))
+      intersect(deps, self$index$id) %||% character(0)
     }
   ),
 
   private = list(
-    #' @field index_unfiltered The unfiltered unscoped index of all packets
+    # The unfiltered unscoped index of all packets
     index_unfiltered = NULL,
-    #' @field index_unfiltered The unfiltered index but scoped index of packets
-    index_scoped = NULL
+    # The unfiltered index but scoped index of packets
+    index_scoped = NULL,
+
+    get_all_packet_depends = function(id, depth) {
+      if (depth == 0) {
```

Comment:
```suggestion
      if (depth <= 0) {
```

Obviously will never happen, but avoids some hard-to-diagnose bugs later when we make a mistake